Mathlib/AlgebraicGeometry/PrimeSpectrum/TensorProduct.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.SurjectiveOnStalks
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+
+/-!
+
+# Lemmas regarding the prime spectrum of tensor products
+
+## Main result
+- `PrimeSpectrum.embedding_tensorProductTo_of_surjectiveOnStalks`:
+  If `R →+* T` is surjective on stalks (see Mathlib/RingTheory/SurjectiveOnStalks.lean),
+  then `Spec(S ⊗[R] T) → Spec S × Spec T` is a topological embedding
+  (where `Spec S × Spec T` is the cartesian product with the product topology).
+-/
+
+variable (R S T) [CommRing R] [CommRing S] [Algebra R S]
+variable [CommRing T] [Algebra R T] [Algebra S T] [IsScalarTower R S T]
+
+open TensorProduct
+
+/-- The canonical map from `Spec(S ⊗[R] T)` to the cartesian product `Spec S × Spec T`. -/
+noncomputable
+def PrimeSpectrum.tensorProductTo (x : PrimeSpectrum (S ⊗[R] T)) :
+    PrimeSpectrum S × PrimeSpectrum T :=
+  ⟨comap (algebraMap _ _) x, comap Algebra.TensorProduct.includeRight.toRingHom x⟩
+
+lemma PrimeSpectrum.continuous_tensorProductTo : Continuous (tensorProductTo R S T) :=
+  (comap _).2.prod_mk (comap _).2
+
+variable (hRT : (algebraMap R T).SurjectiveOnStalks)
+
+lemma PrimeSpectrum.embedding_tensorProductTo_of_surjectiveOnStalks_aux
```

Comment:
`SurjectiveOnStalks` uses `(I : Ideal _) (_ : IsPrime I)`.  Here, you prefer `p : PrimeSpectrum _`. In what case do you prefer which?

Reply:
I only use `PrimeSpectrum` when I need the zarkisi topology.

Reply:
Is it needed here?

Reply:
Oh, I see. The `_aux` function proves half of injectivity, but  embedding is both injectivity and something about the topology.